sys/include/phydat.h

```diff
@@ -178,6 +178,38 @@ const char *phydat_unit_to_str(uint8_t unit);
  */
 char phydat_prefix_from_scale(int8_t scale);
 
+/**
+ * @brief   Scale an integer value to fit into a phydat_t
+ *
+ * Fit an integer value in an existing phydat_t by rescaling all numbers and
+ * updating the scale factor. The result will be rounded towards zero (the
+ * standard C99 integer division behaviour).
+ * The final parameter @p prescale can be used to chain multiple calls to
+ * this function in order to fit multidimensional values into the same phydat_t.
+ *
+ * The code example below shows how to chain multiple calls via the @p prescale parameter
+ *
+ * @code
+ * long val0 = 100000;
+ * long val1 = 2000000;
+ * long val2 = 30000000;
+ * phydat_t dat;
+ * dat.scale = 0;
+ * phydat_fit(&dat, val0, 0, phydat_fit(&dat, val1, 1, phydat_fit(&dat, val2, 2, 0)));
+ * @endcode
```

Comment:
I think markdown syntax

1. looks nicer and
2. allows for syntax highlighting:

```
 * ~~~~~~~~~~~~~~~~~~~~ {.c}
 * long val0 = 100000;
 * long val1 = 2000000;
 * long val2 = 30000000;
 * phydat_t dat;
 * dat.scale = 0;
 * phydat_fit(&dat, val0, 0, phydat_fit(&dat, val1, 1, phydat_fit(&dat, val2, 2, 0)));
 * ~~~~~~~~~~~~~~~~~~~~
```

(note that with doxygen only `~~~~~~~~~~~~` is allowed and that language is determined by file extension (`{.c}`).

Reply:
I didn't get any syntax highlighting regardless of what it put in the curly braces. The docs for Doxygen `@code` command states

https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdcode
>By default the language that is assumed for syntax highlighting is based on the location where the \code block was found. If this part of a Python file for instance, the syntax highlight will be done according to the Python syntax.

So it should be assuming C code here, but I can't get any syntax highlights in my local build.

Reply:
@gebart you don't agree?

Reply:
I tried with the markdown tildes but it looks exactly the same in the generated file

Reply:
Maybe add some more tildes? It works in [`sock`](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/sock/udp.h#L25-L59) (see [rendered results](http://api.riot-os.org/group__net__sock__udp.html)).

Reply:
no difference, I do get the sock_udp example though. I think it may be because of the invalid code (statements outside of a function)

Reply:
But then the code-fragment wouldn't render also, wouldn't it? Maybe remove the `{.c}`? You would lose syntax highlighting, but as I said, I think the fenced style looks better on the eye, when reading the comment verbatim.

Reply:
I guess it is a subjective choice which one is preferred in the code comment but I don't have a strong opinion here, I have updated the PR.